build/builder.go

```diff
@@ -689,7 +719,14 @@ func (b *Builder) Finish() error {
 	return b.buildError
 }
 
-func compareBranches(a, b []zoekt.RepositoryBranch) IndexState {
+// CompareBranches returns an IndexState comparing the two zoekt.RepositoryBranch
+// slices.
+//
+// The result will be IndexStateEqual if both slices specify the same set
+// of branch names and versions, IndexStateBranchSet if either slice specifies a different
+// set of branch names than the other, or IndexStateBranchVersion if both slices specify the same set of
+// branch names but have different accompanying versions.
+func CompareBranches(a, b []zoekt.RepositoryBranch) IndexState {
```

Comment:
I am not sure I agree. I think gitindex is the most obvious call site and we can remove the call site from IndexState. Currently we don't switch based on the return state anyway.

Reply:
Good point! we no longer need to distingiush these values in IndexState. I now agree with you, and additionally we should get rid of the extra IndexState states we added.

Reply:
I'm a bit confused by the is comment chain. As keegan mentioned in https://github.com/sourcegraph/zoekt/pull/310#discussion_r835031449, there are many places that check for this information to pickup inconsistencies. This function is exported because these places exist both inside and outside the build package.

Are we okay in not enforcing invariants in multiple locations?

Reply:
Lets follow-up when we pair later today and record the decision here :)